pkg/provider/vault/vault.go

```diff
@@ -530,3 +551,101 @@ func (v *client) requestTokenWithJwtAuth(ctx context.Context, client Client, jwt
 
 	return token, nil
 }
+
+func (v *client) requestTokenWithCertAuth(ctx context.Context, client Client, certAuth *esv1alpha1.VaultCertAuth, cfg *vault.Config) (string, error) {
+
+	clientKey, err := v.secretKeyRef(ctx, &certAuth.SecretRef)
+	if err != nil {
+		return "", err
+	}
+
+	clientKeyPath, err := getCertPath(clientKey, "client.key")
+	if err != nil {
+		return "", fmt.Errorf(errGetCertPath, err)
+	}
+
+	clientCertPath, err := getCertPath(certAuth.ClientCert, "client.crt")
+	if err != nil {
+		return "", fmt.Errorf(errGetCertPath, err)
+	}
+
+	caCertPath, err := getCertPath(certAuth.CACert, "ca.crt")
+	if err != nil {
+		return "", fmt.Errorf(errGetCertPath, err)
+	}
+
+	tlscfg := vault.TLSConfig{
+		ClientCert: clientCertPath,
+		ClientKey:  clientKeyPath,
+		CACert:     caCertPath,
+	}
```

Comment:
I think we can skip storing the certificates on disk and modify the `http.Client`'s TLSClientConfig directly (like the CABundle implementation above). That's what the `cfg.ConfigureTLS()` does under the hood for you.

https://github.com/hashicorp/vault/blob/api/v1.1.1/api/client.go#L241-L246

Reply:
100%